HttpUtils.cpp

```diff
@@ -230,23 +230,23 @@ HttpUtils::GetStreamHeader(BUrl url, BHttpHeaders* responseHeaders)
 	status_t status;
 	HttpIOReader reader(NULL, responseHeaders, 0);
 
-	HttpRequest request(url, false, "HTTP", &reader);
-	request.SetMethod("GET");
-	request.SetAutoReferrer(true);
-	request.SetFollowLocation(true);
-	request.SetTimeout(10000);
-	request.SetDiscardData(true);
-	request.SetUserAgent("StreamRadio/0.0.4");
+	BHttpRequest* request = dynamic_cast<BHttpRequest*>(BUrlProtocolRoster::MakeRequest(url.UrlString().String(), NULL));
```

Comment:
you are missing a delete request; at the end of the function, otherwise this will leak memory.